### ISSUE TITLE
[BUG] - (Unknown Processor Request Submitted: Palette)

### ISSUE THREAD

**Issue by pixelglade** [tannerhelland/PhotoDemon#571]
PhotoDemon version 8.4

Windows 11.

I have a macro I made in a previous version of PhotoDemon which would change the colour palette of the image from a palette file from GIMP. However in the latest version of PhotoDemon the abililty to change the Palette of an image seems to have been removed from the program and the macro won't work.

When selecting and running the macro from the batch menu, It gives an error: Unknown Processor Request Submitted: Palette

Was the feature moved to somewhere else?

**Comment by tannerhelland** [comment]
Hello, @pixelglade .  I apologize for breaking your Palette macro.

Looking through the PhotoDemon feature history, it looks like the Palette feature was renamed from "Palettize" to "Palette" in March 2022 to better match other photo editors' names for this tool.  This would have broken macros using the old name.

I have added a workaround that maps the old "Palettize" command to the new "Palette" one, but I'm afraid I don't have an immediate way to test the fix because I don't have any old Palettize macros on hand.

At your convenience, could I ask you to try your macro on the latest PhotoDemon nightly build (available [here](https://photodemon.org/download/#nightly)) to let me know if it works now?

Thank you in advance, and I'm sorry again for breaking it!

**Comment by pixelglade** [comment]
Terribly sorry but it seems I made a mistake. It didn't realize I have two copies of Photodemon installed on my computer (9.0 and 8.4). I made the macro in 9.0 and all the paths were set to link to the palette file but I don't think this was the case for the 8.4 version since the palette files were somewhere else. So I think the error was related to not finding the file (I think). I have been able to get the macro file to work in 9.0.

Still, if this means someone using a previous version of the program might not have their script break then the improvement can only be a good thing.
